test/jdk/java/foreign/TestMemorySession.java

```diff
@@ -249,83 +224,25 @@ public void testCloseConfinedLock() {
         }
     }
 
-    @Test(dataProvider = "sessions")
-    public void testSessionAcquires(Supplier<MemorySession> sessionFactory) {
-        MemorySession session = sessionFactory.get();
+    @Test(dataProvider = "allSessions")
+    public void testSessionAcquires(SessionSupplier sessionSupplier) {
+        MemorySession session = sessionSupplier.get();
         acquireRecursive(session, 5);
-        if (session.isCloseable()) {
-            session.close();
-        }
+        if (!SessionSupplier.isImplicit(session))
+            SessionSupplier.close(session);
     }
 
     private void acquireRecursive(MemorySession session, int acquireCount) {
-        try (MemorySession handle = MemorySession.openConfined()) {
-            keepAlive(handle, session);
+        try (Arena arena = Arena.openConfined()) {
+            keepAlive(arena.session(), session);
             if (acquireCount > 0) {
                 // recursive acquire
                 acquireRecursive(session, acquireCount - 1);
             }
-            if (session.isCloseable()) {
-                assertThrows(IllegalStateException.class, session::close);
-            }
-        }
-    }
-
-    @Test
-    public void testConfinedSessionWithImplicitDependency() {
```

Comment:
Is it not possible to keep these 2 tests? Might be nice as a stress test for a naked acquire.